test/coupon_code_test.rb

```diff
@@ -76,23 +76,33 @@ def generate(*args)
     end
 
     it 'should valid code-pretest.' do
-      CouponCode.validate('1K7Q', 1).wont_be_nil
-      CouponCode.validate('1K7C', 1).must_be_nil
+      CouponCode.validate('1K7Q', parts: 1).wont_be_nil
+      CouponCode.validate('1K7C', parts: 1).must_be_nil
 
-      CouponCode.validate('1K7Q-CTFM', 2).wont_be_nil
-      CouponCode.validate('1K7Q-CTFW', 2).must_be_nil
+      CouponCode.validate('1K7Q-CTFM', parts: 2).wont_be_nil
+      CouponCode.validate('1K7Q-CTFW', parts: 2).must_be_nil
 
-      CouponCode.validate('1K7Q-CTFM-LMTC', 3).wont_be_nil
-      CouponCode.validate('1K7Q-CTFM-LMT1', 3).must_be_nil
+      CouponCode.validate('1K7Q-CTFM-LMTC', parts: 3).wont_be_nil
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.